packages/common/src/hardhat/tasks/bridge.js

```diff
@@ -1,27 +1,48 @@
 const { task, types } = require("hardhat/config");
-const { ZERO_ADDRESS } = require("../../Constants");
 
 // Use BLANK_FUNCTION_SIG if you don't want `deposit` or `executeProposal` to delegate a contract call.
 const BLANK_FUNCTION_SIG = "0x00000000";
 // Admin role on Bridge contract:
 const DEFAULT_ADMIN_ROLE = "0x0000000000000000000000000000000000000000000000000000000000000000";
 
 task("register-generic-resource", "Admin can set generic resource ID on Bridge")
-  .addParam("id", "Resource ID to set", "0x0", types.string)
-  .addParam("target", "Contract to delegate call to for this resource ID", ZERO_ADDRESS, types.string)
+  .addParam("target", "Contract to delegate call to for this resource ID", undefined, types.string)
+  .addOptionalParam("rname", "Resouce name to generate Resource ID", undefined, types.string)
+  .addOptionalParam("cid", "Chain ID to generate Resource ID", undefined, types.int)
```

Comment:
This is a cool feature. So example call might be to use `--rname Oracle --cid 1` for any contract communicating with the Source on mainnet?

Reply:
Exactly. I thought forcing them to compute the resource id was cumbersome.

Reply:
cool lgtm